### ISSUE TITLE
Scalable method for adding device commands

### ISSUE THREAD

**Issue by t-sasatani** [Aharoni-Lab/miniscope-io#70]
https://github.com/Aharoni-Lab/miniscope-io/pull/48 added downlink commands, but the command set is still hard-coded and complicated to expand. We should unify the command set (related: https://github.com/Aharoni-Lab/miniscope-io/pull/48#discussion_r1830337498), or maybe feed them using the device config. Using device config might be better because the available peripherals and bounds of the values vary between devices, which seems relatively easy to handle with a nested YAML file.

**Comment by sneakers-the-rat** [comment]
Agreed. This is part of the API refactor (which im realizing I should have done separately from the pipelining system, but so it goes) - each device will have a `configure` method that takes an instance of that device's `DeviceConfig`. So we will want to do a few things to make that possible, also described in other issues
- make a user config directory
- commands for generating/copying default configs as a starting point that can then be modified
- IDs for configs so we can do something like `mio configure MyDevice` rather than needing to refer to a path

Then we can also have cli for setting specific options like `mio configure MyDevice fps=30` or whatever that both sets the value on the device and sets it in the config file so they stay sync'd
